larpandora/LArPandoraInterface/LArPandoraHelper.cxx

```diff
@@ -1500,6 +1500,17 @@ namespace lar_pandora {
     return larpandoraobj::PFParticleMetadata(pPfo->GetPropertiesMap());
   }
 
+  bool
+  LArPandoraHelper::IsDualPhase()
+  {
+    art::ServiceHandle<geo::Geometry const> theGeometry;
+    std::unordered_set<geo::_plane_sigtype> sigtypeSet;
+    for (readout::ROPID const& rID: theGeometry->IterateROPIDs()) {
+      sigtypeSet.insert(theGeometry->SignalType(rID));
+    }
+    return ((!sigtypeSet.count(geo::kInduction) || !sigtypeSet.count(geo::kCollection)) && !sigtypeSet.count(geo::kMysteryType));
```

Comment:
What actually **is** the signal type for dual phase here? Also, I wonder if it's worthwhile to split this line into a few descriptive steps to make it a little easier to process. Not sure about everyone else but I find the not(A) or not(B) where A and B involve additional function calls harder to process than say !(hasInduction && hasCollection).

Reply:
Yeah, will replace with !(hasInduction && hasCollection). The signal type can be one of the three: kInduction, kCollection and kMystery. I think we should reject the mystery case.

Reply:
Ok, so there's no specific type for dual phase then, it's really just confirming that there is one, and only one of the types (and yes, definitely agree on rejecting the mystery case, I was just curious as to whether or not another specification existed).

Reply:
Yes, unfortunately that's the case. This check is not necessarily robust against other potential single-readout-type non-DP future detectors, but works for now.

Reply:
Possibly worthy of an ATTN comment here in that case? Along with a note on what exactly is done for the known dual phase case?

Reply:
Changed the logic according to your suggestion!